flecsi-sp/burton/burton_specialization_init.cc

```diff
@@ -61,16 +64,34 @@ void specialization_tlt_init(int argc, char** argv)
 
   //===========================================================================
   // Partition mesh
-  //===========================================================================
+  //===========================================================================  
 
   clog(info) << "Partitioning mesh" << std::endl;
 
   // need to put the filename into a statically sized character array
   auto mesh_filename = flecsi_sp::utils::to_char_array( mesh_filename_string );
 
+  // io definition params
+  // using real_t = flecsi_sp::burton::burton_mesh_t::real_t;
+  // constexpr auto num_dims = flecsi_sp::burton::burton_mesh_t::num_dimensions;
+  // using exo_def_t = flecsi_sp::io::exodus_definition__<num_dims, real_t>;
+  // using mpas_def_t = flecsi_sp::io::mpas_definition_u<real_t>;
+
+  auto extension = ristra::utils::file_extension(mesh_filename_string);
+
   // execute the mpi task to partition the mesh
-  flecsi_execute_mpi_task(partition_mesh, flecsi_sp::burton, mesh_filename,
-    max_entries);
+  if(extension == "exo" || extension == "g") {
```

Comment:
@ipdemes @dmringo I'm looking at extending our runtime selection of type of mesh to parse, in particular looking towards X3D format meshes.  Is having separate tasks for each format the approach you two have decided upon?  I had stubbed out some code that pushed the `if` check down into the task (the singular `partition_mesh` task), but this involved casting the mesh definition object (e.g. `exodus_definition_t` or `x3d_definition_t`) to the common base class `mesh_definition_u` for some operations.  That was part of the motivation of [PR 553 in FleCSI](https://github.com/laristra/flecsi/pull/553)

If we want to go the route of separate tasks for partitioning and initializing, that is fine.  It cleans up some things, but does require, perhaps, a bit of code duplication.  In general I'm against that sort of thing, but in this particular case I don't really have strong feelings either way.

Reply:
@cmsquared : do you have your work in a branch somewhere? So I can have a look?

Reply:
Unfortunately not.  But the changes to the `partition_mesh` task looked something like this:
```c++
  // mesh definitions
  using mesh_definition_t = flecsi::topology::mesh_definition_u<num_dims>;
  using exodus_definition_t = flecsi_sp::io::exodus_definition__<num_dims, real_t>;
  using x3d_definition_t = flecsi_sp::io::x3d_definition__<num_dims, real_t>;

  // load the mesh
  auto filename_string = filename.str();
  std::unique_ptr<mesh_definition_t> mesh_def;
  if (filename_string.find(".exo") != std::string::npos)
    mesh_def = std::make_unique<exodus_definition_t>(filename_string);
  else if (filename_string.find(".x3d") != std::string::npos)
    mesh_def = std::make_unique<x3d_definition_t>(filename_string);
  else
    clog_fatal("Burton specialization doesn't know how to read " <<
               filename_string);
```

Reply:
@cmsquared I'm looking at this again, and liked your approach, but ran into problems resolving the `write` method on the mesh definition (called near the end of `partition_mesh`) since it's not part of the abstract `mesh_definition_u` interface.  Did you cast to the concrete implementation to solve this?  Unless there are plans to add that method to the interface in FleCSI, that makes it seem like a `partition_mesh` task templated on the IO backend would be preferred.